mpd/mpd.go

```diff
@@ -219,21 +255,24 @@ func (period *Period) SetDuration(d time.Duration) {
 // segmentAlignment - Segment Alignment(i.e. true).
 // startWithSAP - Starts With SAP (i.e. 1).
 // lang - Language (i.e. en).
-func (m *MPD) AddNewAdaptationSetAudio(mimeType string, segmentAlignment bool, startWithSAP int64, lang string) (*AdaptationSet, error) {
-	return m.period.AddNewAdaptationSetAudio(mimeType, segmentAlignment, startWithSAP, lang)
+func (m *MPD) AddNewAdaptationSetAudio(id string, mimeType string, segmentAlignment bool, startWithSAP int64, lang string) (*AdaptationSet, error) {
+	return m.period.AddNewAdaptationSetAudio(id, mimeType, segmentAlignment, startWithSAP, lang)
 }
 
 // Create a new Adaptation Set for Audio Assets.
 // mimeType - MIME Type (i.e. audio/mp4).
 // segmentAlignment - Segment Alignment(i.e. true).
 // startWithSAP - Starts With SAP (i.e. 1).
 // lang - Language (i.e. en).
-func (period *Period) AddNewAdaptationSetAudio(mimeType string, segmentAlignment bool, startWithSAP int64, lang string) (*AdaptationSet, error) {
+func (period *Period) AddNewAdaptationSetAudio(id string, mimeType string, segmentAlignment bool, startWithSAP int64, lang string) (*AdaptationSet, error) {
```

Comment:
This is effectively making `id` a required field now, what's the reason for this?

Reply:
@mbowBC ^

Reply:
@id is mandatory on dynamic streams (I think)
AdaptationSet@id is an optional parameter
https://github.com/bbcrddave/dash.js/commit/70439223aea8f7aabefddd500a49e3d17f36b12a
and was found from http://standards.iso.org/ittf/PubliclyAvailableStandards/MPEG-DASH_schema_files/DASH-MPD.xsd
search for 
<xs:attribute name="id" type="xs:unsignedInt"/>